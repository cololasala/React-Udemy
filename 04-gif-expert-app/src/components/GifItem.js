import React from 'react';
import PropTypes from 'prop-types';

const GifItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__fadeInDown">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}


export default GifItem;