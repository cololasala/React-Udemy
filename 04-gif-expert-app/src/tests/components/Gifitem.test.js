import { shallow } from 'enzyme';
import GifItem from '../../components/GifItem';

describe('Prueba de componente GifItem.js', () => {

    const title = 'Un titulo';
    const url = 'http://localhost/algo.jpg'
    const wrapper = shallow(<GifItem title={title} url={url} />);

    test('debe mostrar <GifItem/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    });

    test('debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        // console.log(img.html());             puedo ver como es la etiqueta con sus propiedades
        // console.log(img.props())             puedo ver como son las propiedades de la etiqueta (es mejor usar esto)

        expect(img.prop('src')).toBe(url);      // img.prop('src') obtengo el valor en el src
    });

    test('div debe tener la clase animate__fadeInDown', () => {
        const div = wrapper.find('div');
        const classAnimate = 'animate__fadeInDown';
       // console.log(div.prop('className').split(' ')[2]);

      //  expect(div.prop('className').split(' ')[2]).toBe(classAnimate); esta es otra manera de hacerlo

        expect(div.prop('className').includes(classAnimate)).toBe(true);
    })
})