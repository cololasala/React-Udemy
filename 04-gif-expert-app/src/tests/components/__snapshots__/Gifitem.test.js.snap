// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente GifItem.js debe mostrar <GifItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
